File: backend/src/main/java/at/ac/tuwien/mogda/willgraph/controller/dto/ListingSearchFilterDto.java
package at.ac.tuwien.mogda.willgraph.controller.dto;

import lombok.Data;

import java.util.List;

@Data
public class ListingSearchFilterDto {
    private ListingCriteria listing;
    private TransportCriteria transport;
    private List<PriorityItemDto> amenityPriorities;
    private List<PriorityItemDto> poiPriorities;
    private Double bonusScoreRegion = 0.0;
}

File: webapp/src/types/dto/ListingSearchFilterDto.ts
import type { ListingCriteria } from "./ListingCriteria";
import type { TransportCriteria } from "./TransportCriteria";
import type { PriorityItemDto } from "./PriorityItemDto";

export interface ListingSearchFilterDto {
	listing: ListingCriteria | null;
	transport: TransportCriteria | null;
	amenityPriorities: PriorityItemDto[] | null;
	poiPriorities: PriorityItemDto[] | null;
}
